### DISCUSSION TITLE
Variable error on launch

### DISCUSSION THREAD

**Discussion by Youngwhisk** [RomanKubiak/ctrlr#249] [General]
I have this as part of a method that is called on a change event :

for c = 2, 3 do
panel:getModulatorByName("modulator-"..c):getComponent():setProperty("componentEffectRadius",0, true)
end
panel:getModulatorByName("modulator-"..testgroup):getComponent():setProperty("componentEffectRadius",4, true)
end

It throws an error on launch saying that trying to concatenate fails because the global variable "testgroup" has a nil value. However, if I compile the method after launch there is no error and everything works properly. The "testgroup" variable is set by the choice of a radio button so I can't give it a value in this method.

Is there a way to initialize it elsewhere?

Actually an if statement working on either a 0 or 1 value of the radio button would be ideal but I cant seem to get it to work inside a change method.

I will have 15 different groups doing this. All this does is set the glow range on the selected radio button in a group. It works very well. I am not aware of another way to do this.

Thanks

**Comment by Ted-juh** [comment]
I think it cannot concatenate because the "c" from the for loop is outside the string to build the name of the component you want to reach. To get to "modulator-2" or "modulator-3" you might need to do:

for c = 2,3 do
panel:getModulatorByName(string.format("modulator-%s", c)):getComponent():setProperty("componentEffectRadius,0,true)
end

You can initialize by creating a init file and attach it to "luaPanelLoaded" or "called when the panel finished loading" by clicking at an empty space in your panel and search in the General properties pane in editor mode.

And what is it that you cannot get to work about the "if" statement?

radioButtonValue = panel:getModulatorByName("modulatorName"):getValue()
if radioButtonValue == 0 then
-- do some glowy stuff
else
-- do not dare to do glowy stuff
end

**Reply by Youngwhisk** [reply]
I'm not able to look at it right now but it didn't seem concerned with the c variable, just the testgroup variable and the testgroup variable is apparently initialized after the error check.

I will try the if statement again. I like it better. 

Then the initializing if needed.

**Reply by dnaldoog** [reply]
Try putting this as the first line in your function

if source == 6 or source == 1 then return end

**Reply by Youngwhisk** [reply]
I was trying to be too clever. None of the above worked. In the If statement inside the For loop (using a change method), every modulator value showed as 1, never 0. I finally realized that I could just put the For statement into my radio button mouse down method and hard code the fourth line variable into a modulator name and it all works fine. It also gives me individual control over the glow radius for each radio button depending on label length.

Thanks for the advice.
